var/spack/repos/builtin/packages/py-cylc-flow/package.py

```diff
@@ -0,0 +1,36 @@
+# Copyright 2013-2023 Lawrence Livermore National Security, LLC and other
+# Spack Project Developers. See the top-level COPYRIGHT file for details.
+#
+# SPDX-License-Identifier: (Apache-2.0 OR MIT)
+
+from spack.package import *
+
+
+class PyCylcFlow(PythonPackage):
+    """A workflow engine for cycling systems."""
+
+    homepage = "https://cylc.org"
+    pypi = "cylc-flow/cylc-flow-8.1.4.tar.gz"
+
+    maintainers("LydDeb")
+
+    version("8.2.0", sha256="cbe35e0d72d1ca36f28a4cebe9b9040a3445a74253bc94051a3c906cf179ded0")
+    version("8.1.4", sha256="d1835ac18f6f24f3115c56b2bc821185484e834a86b12fd0033ff7e4dc3c1f63")
+
+    depends_on("py-setuptools@49:66,68:", type=("build"))
```

Comment:
Thank you for this proofreading. I stupidly removed the "run" because I'm used to only seeing "build" for the py-setuptools type.

Out of curiosity, I quickly compared the occurrences of "type=.*when=" and "when=.*type=":
```
$ grep -l "depends_on(.*when=.*type=" var/spack/repos/builtin/packages/*/package.py | wc -l
813
$ grep -l "depends_on(.*type=.*when=" var/spack/repos/builtin/packages/*/package.py | wc -l
1258
$ grep "depends_on(.*when=.*type=" var/spack/repos/builtin/packages/*/package.py | wc -l
4160
$ grep "depends_on(.*type=.*when=" var/spack/repos/builtin/packages/*/package.py | wc -l
5371
```
It seems that there is a secret war between typewhen and whentype and that whentype contributed 1.5 times more than typewhen.

Likewise if we only look at the python package recipes:
```
$ grep -l "depends_on(.*when=.*type=" var/spack/repos/builtin/packages/py-*/package.py | wc -l
555
$ grep -l "depends_on(.*type=.*when=" var/spack/repos/builtin/packages/py-*/package.py | wc -l
319
```

Reply:
typewhen used to be more popular. I started using whentype and convinced @manuelakuhn to follow me. My only reason is that this is the order they are defined in the definition of the `depends_on` directive. Nothing breaks if you use a different order. We should really mass-convert all of them to one order or the other.